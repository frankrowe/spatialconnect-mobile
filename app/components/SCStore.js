import React, { PropTypes } from 'react';
import {
  ScrollView,
} from 'react-native';
import Property from './Property';
import { propertyListStyles } from '../style/style';

const SCStore = props => (
  <ScrollView style={propertyListStyles.container}>
    <Property name={'Name'} value={props.storeInfo.name} />
    <Property name={'Type'} value={props.storeInfo.type} />
    <Property name={'ID'} value={props.storeInfo.storeId} />
    {props.storeInfo.vectorLayers && props.storeInfo.vectorLayers.length ?
      <Property name={'Vector Layers'} value={props.storeInfo.vectorLayers.join('\n')} />
    : null}
    {props.storeInfo.rasterLayers && props.storeInfo.rasterLayers.length ?
      <Property name={'Raster Layers'} value={props.storeInfo.rasterLayers.join('\n')} />
    : null}
  </ScrollView>
);

SCStore.propTypes = {
  storeInfo: PropTypes.object.isRequired,
};

export default SCStore;
