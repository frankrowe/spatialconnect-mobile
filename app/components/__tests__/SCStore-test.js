import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import stores from 'stores';
import SCStore from '../SCStore';

describe('<SCStore />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <SCStore storeInfo={stores[0]} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
