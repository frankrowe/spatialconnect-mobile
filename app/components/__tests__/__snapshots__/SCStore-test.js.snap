exports[`<SCStore /> renders correctly 1`] = `
<ScrollView
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
      "marginTop": -1,
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flexDirection": "column",
      }
    }>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#f0f1f4",
          "borderBottomWidth": 0,
          "borderColor": "#dddddd",
          "borderTopWidth": 0,
          "justifyContent": "center",
          "padding": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }>
        Name
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
          "padding": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 12,
          }
        }>
        gj1
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flexDirection": "column",
      }
    }>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#f0f1f4",
          "borderBottomWidth": 0,
          "borderColor": "#dddddd",
          "borderTopWidth": 0,
          "justifyContent": "center",
          "padding": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }>
        Type
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
          "padding": 10,
        }
      } />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flexDirection": "column",
      }
    }>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#f0f1f4",
          "borderBottomWidth": 0,
          "borderColor": "#dddddd",
          "borderTopWidth": 0,
          "justifyContent": "center",
          "padding": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }>
        ID
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
          "padding": 10,
        }
      } />
  </View>
</ScrollView>
`;
